lib/ecto/changeset.ex

```diff
@@ -236,6 +236,34 @@ defmodule Ecto.Changeset do
         |> Ecto.Changeset.validate_required(...)
         |> Ecto.Changeset.validate_length(...)
 
+  Besides the basic types which are mentioned above, such as `:boolean` and `:string`, 
+  parameterized types can alse be used in schemaless changesets. It's just that parameterized
+  types require a three-element tuple to work:
+
+      {:parameterized, typemod, opts}
+
+  The first element `:parameterized` indicates you want to use a parameterized type.
+  The second element indicates which module to use for handling data. The third element 
+  is the options for this type. 
+
+  Generally, parameterized types will implement `Ecto.ParameterizedType` behaviour. Because
+  of that, in actual usage, we call `Ecto.ParameterizedType.init/2` directly, which will 
+  generate the three-element tuple we need.
+
+  Take `Ecto.Enum`, which implements `Ecto.ParameterizedType`, as an example:
+
+      data  = %{}
+      types = %{
+        name: :string, 
+        sex: Ecto.ParameterizedType.init(Ecto.Enum, values: [:male, :female, :secret])
+      }
+      params = %{name: "Callum", sex: "male"}
+      changeset =
+        {data, types}
+        |> Ecto.Changeset.cast(params, Map.keys(types))
+        |> Ecto.Changeset.validate_required(...)
+        |> Ecto.Changeset.validate_length(...)
```

Comment:
```suggestion
      types = %{
        name: :string, 
        role: Ecto.Enum.init(values: [:reader, :editor, :admin])
      }

      data  = %{}
      params = %{name: "Callum", role: "reader"}

      changeset =
        {data, types}
        |> Ecto.Changeset.cast(params, Map.keys(types))
        |> Ecto.Changeset.validate_required(...)
        |> Ecto.Changeset.validate_length(...)
```

Reply:
I found I have to write something like this:
```diff
- role: Ecto.Enum.init(values: [:reader, :editor, :admin])
+ role: Ecto.ParameterizedType.init(Ecto.Enum, values: [:reader, :editor, :admin])
```